src/main/scala/com/amazon/deequ/analyzers/Compliance.scala

```diff
@@ -34,8 +35,9 @@ import com.google.common.annotations.VisibleForTesting
   *                         describing what the analysis being done for.
   * @param predicate SQL-predicate to apply per row
   * @param where Additional filter to apply before the analyzer is run.
+  * @param columns List of columns used for predicate - This is needed to run pre condition check!
   */
-case class Compliance(instance: String, predicate: String, where: Option[String] = None)
+case class Compliance(instance: String, predicate: String, columns: List[String], where: Option[String] = None)
```

Comment:
The new argument `columns` should be optional for backwards-compatibility, since users didn't have to specify it until now for the analyzer to work. Most people will only instantiate a `Compliance` object from a Check, so your code should still work as is, but since `Compliance` is a public class we should ensure it can be constructed without the column list.

Reply:
Also, could this be a trait? That way you don't have to implement `additionalPreconditions` here, you just rely on the trait (something like `RequiresColumnsToExist`) to define the preconditions, and other checks can inherit it effortlessly

More of a matter of opinion - if you see it works better as it is right now I have no objections.

Reply:
I tried to make it a trait, changes felt not future proof. Following the pattern of `additionalPreconditions` as this function can add additional checks later.
Example: one from [completeness](https://github.com/awslabs/deequ/blob/master/src/main/scala/com/amazon/deequ/analyzers/Completeness.scala#L47)

I have made updates to make `columns` as `Option` and added the tests. Will update the PR once the internal review is complete.